### ISSUE TITLE
make module-init having problem with remote repositories

### ISSUE THREAD

**Issue by mattcary** [kubernetes/website#30608]
**Problem:**

`make module-init` seems to have problems with my remotes---more accurately, `git submodules` does. Running `make module-init` produces

```
Initializing submodules...
remote: Total 0 (delta 0), reused 0 (delta 0), pack-reused 0        
fatal: 'upstream' does not appear to be a git repository
fatal: Could not read from remote repository.

Please make sure you have the correct access rights
and the repository exists.
fatal: Fetched in submodule path 'api-ref-generator', but it did not contain 55bce686224caba37f93e1e1eb53c0c9fc104ed4. Direct fetching of that commit failed.
fatal: 
make: *** [Makefile:27: module-init] Error 128
```

My remotes are setup like

```
$ git remote -v
fork	git@github.com:mattcary/website.git (fetch)
fork	git@github.com:mattcary/website.git (push)
origin	git@github.com:kubernetes/website.git (fetch)
origin	git@github.com:kubernetes/website.git (push)
upstream	git@github.com:kubernetes/website.git (fetch)
upstream	git@github.com:kubernetes/website.git (push)

$ git config -l | grep defaultremote
checkout.defaultremote=origin
```

I tried to manually run git submodule commands, but apparently git is smarter than I am and nothing is working.

**Comment by sftim** [comment]
I just tested this and it does work on my machine
Try switching to the `main` branch and running that command once again @mattcary  (stash or commit your work first)

If that fails: `rm -rf api-ref-generator` (from the top of the repo) and run, manually:
```bash
git submodule update --init --recursive
```

**Comment by sftim** [comment]
/kind support

I think - happy to offer support here, it's a reasonable thing to want help with
/remove-kind bug

**Comment by sftim** [comment]
Did that advice help you troubleshoot things @mattcary?

**Comment by mattcary** [comment]
I got something working by removing remotes, fetching, but then had some weird restoring that I had to do in my branch in order to launch the site. So something weird going on, I need start from scratch and work through systematically. I wonder if the problem is because I had two remotes that pointed to the same place (but I'm not sure why git submodules is so persnickity).

I'll post more complete reproduction steps when I have a chance. Thank for your help on this!
